### DISCUSSION TITLE
Which platform support Dreame D9?

### DISCUSSION THREAD

**Discussion by Latyn4ik** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#410] [Q&A] [answered]
Hi i'm trying integrate my Dreame D9 to Home Assistant. I setup map extractor, but i don`t understand how to take vacuum  entity. Which platform support  Dreame D9?
https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card#supported-vacuum-platforms

**Comment by PiotrMachowski** [comment, marked as answer]
Each supported platform has its own documentation with a list of supported vacuums.

**Reply by Latyn4ik** [reply]
@PiotrMachowski I understand that only this platform **send_command** supports Dreame D9, but for this i need to flash it with custom firmware. Is it correct?

**Reply by Latyn4ik** [reply]
XIaomi MIIO don`t support it
![image](https://user-images.githubusercontent.com/25727082/172218141-c02e60db-3aa1-4002-b3c9-92b0451f3291.png)

**Reply by PiotrMachowski** [reply]
@Latyn4ik ValetudoRE is a custom firmware for Xiaomi Vacuums. Platform "send_command" still requires vacuum to be somehow added to HA. You gave to find an integration that works with your model and only then try to add it to this card.

**Reply by PiotrMachowski** [reply]
> XIaomi MIIO don`t support it ![image](https://user-images.githubusercontent.com/25727082/172218141-c02e60db-3aa1-4002-b3c9-92b0451f3291.png)

This is a plug, not a vacuum

**Reply by Latyn4ik** [reply]
@PiotrMachowski thank you for answers.

**Reply by Latyn4ik** [reply]
> > XIaomi MIIO don`t support it ![image](https://user-images.githubusercontent.com/25727082/172218141-c02e60db-3aa1-4002-b3c9-92b0451f3291.png)
> 
> This is a plug, not a vacuum

It`s appeared when u chose vacuum
![image](https://user-images.githubusercontent.com/25727082/172258166-941cc8db-2f58-4838-b89f-2d35689cd149.png)
and after this appeared this one:
![image](https://user-images.githubusercontent.com/25727082/172258237-7267dea8-237d-4694-a14c-6b65c291f1ce.png)

**Reply by PiotrMachowski** [reply]
> It`s appeared when u chose vacuum
> ![image](https://user-images.githubusercontent.com/25727082/172258166-941cc8db-2f58-4838-b89f-2d35689cd149.png)

It looks ok for me. Maybe it has already created an entity? I don't use this integration, so I can't really help
